torch/_dynamo/variables/builder.py

```diff
@@ -503,12 +503,13 @@ def index_source(key):
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
         elif isinstance(value, torch.cuda.streams.Stream):
-            return CUDAStreamVariable(
-                None,
-                value,
-                source=self.source,
-                guards=self.make_guards(GuardBuilder.ID_MATCH),
-            )
+            unimplemented("CUDAStreamVariable does not currently work soundly.")
+            # return CUDAStreamVariable(
+            #     None,
+            #     value,
+            #     source=self.source,
+            #     guards=self.make_guards(GuardBuilder.ID_MATCH),
+            # )
```

Comment:
I think we should identify what caused the error, it seems we missed ```device``` for ```CUDAStreamVariable```, can we add it? If you are urgent to unblock anything but don't want to graph break, you can also try to wrap it as null context manager and print warnings like this:
https://github.com/pytorch/pytorch/blob/970c60b336b7bc346cd1cbb8b86f547983c9668a/torch/_dynamo/variables/torch.py#L382-L389

Reply:
Yanbo, would your suggestion to add device to CudaStreamVariable just paper over the fact that we somehow leaked a VariableTracker object outside of our compiler?  It's important to not let that happen.